src/core/refiner.ml

```diff
@@ -0,0 +1,321 @@
+open Common
+open Lplib
+open Timed
+open Term
+
+let log = Debug.new_logger 'z' "refi" "Refiner"
+let log = log.logger
+
+(** Module that provide a lookup function to the refiner. *)
+module type LOOKUP = sig
+  val lookup : ctxt -> term -> term -> ((term -> term) * constr list) option
+  (** [lookup a b] returns a coercion from [a] to [b] in context [ctx], if it
```

Comment:
[lookup ctx a b]

Could you also explain what is returned?

Reply:
For the moment, `lookup a b` returns a function `f` and constraints `eqs` such that if `t` is a term of type `a`, then `f t` is of type `b` if `eqs` are solvable.

But this behaviour might change.

Reply:
Thanks to add this in the comment.

Reply:
It has been changed, now the coercion returns a Bindlib binder.

Reply:
The comment is still not very clear. Why not saying something like (if I understand correctly): "[lookup a b] returns a pair [Some(coerce,cs)] where [coerce] is a term binder that is a coercion from terms of type [a] to terms of type [b], and [cs] is a list of constraints for [subst coerce (Vari x)] to be of type [b] when [x] is a fresh variable of type [a], if such a coercion exists. It returns [None] otherwise."?